streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskExecutor.java

```diff
@@ -191,43 +191,49 @@ void commitOffsetsOrTransaction(final Map<Task, Map<TopicPartition, OffsetAndMet
                         corruptedTasks.add(task.id());
                     }
                 }
-            } else {
-                final Map<TopicPartition, OffsetAndMetadata> allOffsets = offsetsPerTask.values().stream()
-                    .flatMap(e -> e.entrySet().stream()).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            }
+        } else {
+            final Map<TopicPartition, OffsetAndMetadata> allOffsets = offsetsPerTask.values().stream()
+                .flatMap(e -> e.entrySet().stream()).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+            if (executionMetadata.processingMode() == EXACTLY_ONCE_V2) {
+                if (!offsetsPerTask.isEmpty() || taskManager.threadProducer().transactionInFlight()) {
```

Comment:
nit: 
```suggestion
                if (!allOffsets.isEmpty() || taskManager.threadProducer().transactionInFlight()) {
```

Reply:
While breaking up each processing mode into its own separate branch as you requested, I decided to move the `allOffsets` map inside this if condition -- ie it still uses the `offsetsPerTask` map for the emptiness check. I'm assuming this is fine since it's just a nit but lmk if you still prefer to do the check on the `allOffsets` map instead

Reply:
Yes, moving `allOffsets` inside the branch is fine with me.
However, I actually proposed to extract each branch to a separate method, so that you have something like:

```java
if (executionMetadata.processingMode() == EXACTLY_ONCE_ALPHA) {
    commitForEosv1(...);
} else if (executionMetadata.processingMode() == EXACTLY_ONCE_V2) {
    commitForEosv2(...);
} else {
    commitForAlos(...);
}
```

Reply:
Ah I see. Will do